code-reviews.md

```diff
@@ -0,0 +1,141 @@
+
+### For code submitters
+
+- Link to the code review from the originating Jira task/issue, if applicable.
+- Remember that the code isn't you, don't get defensive when a reviewer is critical
+  of the code.
+- Seek to understand the reviewer's perspective.
+- Try to respond to every comment.
+- Be grateful for the reviewer's suggestions. ("Good catch, fixing in a4994ec")
+- Explain why the code exists. ("We need to work around these existing patterns")
+- Extract out-of-scope changes and refactorings into future Jira tasks/issues.
+- Push commits based on earlier rounds of feedback as isolated commits to the
+  branch. Do not squash until the branch is ready to merge. Reviewers should be
+  able to read individual updates based on their earlier feedback.
+- Wait to merge the branch until it has passed Continuous Integration testing.
+  (TDDium, TravisCI, etc.)
+- Merge once you feel confident in the code and its impact on the project.
+
+### For code reviewers
+
+- Understand why the code is necessary (bug, user experience, refactoring)
+- Seek to understand the author's perspective.
+- Clearly communicate which ideas you feel strongly about and those you don't.
+- Identify ways to simplify the code while still solving the problem.
+- Offer alternative implementations, but assume the author already considered
+  them. ("What do you think about such-and-such here?")
+- If discussions turn too philosophical or academic, move the discussion to a new
+  issue or offline. In the meantime, the author has the final say on the current
+  implementation.
+- Sign off on the pull request with a :thumbsup: or "Ready to merge" comment.
+
+
+## Submitting code
+
+Before seeking a review, you should be able to check off each of the following:
+
+[ ] Changes are limited to a single goal (no scope creep)
+[ ] Code can be automatically merged (no conflicts)
+[ ] Code follows the standards laid out in this playbook
+[ ] Passes all existing automated tests
+[ ] New functions include new tests
+[ ] New functions are documented (with a description, list of inputs, and
+    expected output)
+[ ] Placeholder code is flagged
+[ ] Visually tested in supported browsers and devices
+[ ] Project documentation has been updated (including the "Pending" section of
+    the CHANGELOG)
+
+### Opening a PR
+
+Once sure your code meets the checks above, open a Pull Request for your new
+feature or bugfix following our [PR Template](https://raw.githubusercontent.com/cfpb/front-end/733397192b04551d56c847f3416b0bcdf5bab544/pr-template.md). Remember that the
+more information you provide to reviewers, the more context they will have. This
+leads to faster reviews, and less back and forth between everyone.
+
+### Ready to merge?
+
+You've got your sign-offs, now what? If you've made any review edits, it's best
+to squash them to keep the history clean (though not required). Use the merge button
+on Github (don't merge locally) and ensure any automated deployments succeed
```

Comment:
That's a question I wanted to bring up. Within our team, whoever opens is also the person that closes. We do this for a couple of reasons:

- gives author a chance to cleanup and squashing any edits (if desired);
- fix conflicts;
- merge when appropriate (sometimes multiple PRs need to go together).

So I wasn't sure if we should mandate this here, point out that it's a team-by-team basis, or let it fall where it may.